### ISSUE TITLE
Unexpected unsign passing on a random signature postfix

### ISSUE THREAD

**Issue by muturgan** [tj/node-cookie-signature#40]
Hi dear team.
Thank you for the amazing project!

This is my reproduce example:
``` js
const cookie = require('cookie-signature');
const value = 'value';
const secret = 'secret';
const signed = cookie.sign(value, secret);
const result = cookie.unsign(signed + 'some', secret);
console.log(result)
```

I expected to see a `false` result. But got a `'value'`.

**Comment by natevw** [comment]
Thanks for the report and helpful example!

Just to be clear for anyone else following along, I wouldn't consider this a *vulnerability* since the cookie must still be properly signed for it to happen.

But I agree that this is a regression introduced during [another bugfix](94581e68205ff63a9191312ccf9366089ef960ab). So I think it's safe to assume (for semver purposes) that nobody should be relying on this bug and we can fix it.

It happens in the chain of events where `val` is silently truncated when copied into a buffer for comparison purposes:

```
valBuffer = Buffer.alloc(macBuffer.length);
valBuffer.write(val);
return crypto.timingSafeEqual(macBuffer, valBuffer) ? str : false;
```

We **do** need `valBuffer` to be the right length otherwise `timingSafeEqual` will throw an exception but we should probably check that `.write(val)` was the expected size. This might be a bit tricky w.r.t. UTF-8 encoding, i.e. the expected size is not going to always equal `val.length`! (That situation is hopefully not super common with actual `Cookie` headers but people could be using this algorithm in other contexts where non-ASCII data doesn't need to be handled as carefully.)

**Comment by natevw** [comment]
I've got a proposed fix for this in #41, but am waiting for some other maintainers to sign off on it before merging it in.

If this bug is blocking your own development you could `npm install tj/node-cookie-signature#nvw-unsign-fixes` in the meantime but I suspect most users can wait until the fix is reviewed and published.
